### PR TITLE
subsys: net: Switch to use upstream CoAP

### PR DESCRIPTION
**Pull request by rlubos**
Upstream CoAP library was reworked so it's ready to use with sockets. Therefore we should use the upstream library in the applications, to avoid maintenance burden.

As agreed, the downstream library is removed and reserved for internal use.

Tested the app with nrf9160 and qemu_x86, both seem to work fine.

### CODE REVIEW

File: samples/nrf9160/coap_client/src/main.c [hunk 1]
```diff
@@ -305,12 +279,32 @@ void main(void)
 				continue;
 			}
 
-			return;
+			printk("Poll error, exit...\n");
+			break;;
```
Comment (jtguggedal) [inline samples/nrf9160/coap_client/src/main.c]: Redundant extra `;`

File: samples/nrf9160/coap_client/src/main.c [hunk 2]
```diff
@@ -92,115 +86,98 @@ static int client_init(void)
...
+		printk("Failed to create CoAP socket: %d.\n", errno);
+		return -errno;
 	}
 
-	/* NOTE: transport_handle is the socket descriptor. */
-	transport_handle = local_port_list[0].transport;
+	err = connect(sock, (struct sockaddr *)&server,
+		      sizeof(struct sockaddr_in));
+	if (err < 0) {
+		printk("Connect failed : %d", errno);
```
Comment (jtguggedal) [inline samples/nrf9160/coap_client/src/main.c]: Missing newline

Comment (rlubos) [reply samples/nrf9160/coap_client/src/main.c]: Where exactly?

Comment (jtguggedal) [reply samples/nrf9160/coap_client/src/main.c]: Was thinking about `\n` at the end of the printk message

Comment (rlubos) [reply samples/nrf9160/coap_client/src/main.c]: Ah right, will fix, thanks!

**Pull request merged**